x/gov/types/v1beta2/proposals_test.go

```diff
@@ -23,3 +27,16 @@ func TestProposalStatus_Format(t *testing.T) {
 		require.Equal(t, tt.expectedStringOutput, got)
 	}
 }
+
+// TestNestedAnys tests that we can call .String() on a struct with nested Anys.
+// Here, we're creating a proposal which has a Msg (1st any) with a legacy
+// content (2nd any).
+func TestNestedAnys(t *testing.T) {
+	testProposal := v1beta1.NewTextProposal("Proposal", "testing proposal")
+	msgContent, err := v1beta2.NewLegacyContent(testProposal, "cosmos1govacct")
+	require.NoError(t, err)
+	proposal, err := v1beta2.NewProposal([]sdk.Msg{msgContent}, 1, time.Now(), time.Now())
+	require.NoError(t, err)
+
+	require.Equal(t, "TODO Fix panic here", proposal.String())
+}
```

Comment:
I managed to extract the failing test into a minimal reproducible example.

Basically, when calling `.String()` on a proto struct that has nested Anys, it panics. A quick debugging leads to the field `any.cachedValue` panicking.

Any ideas on how to solve this? One way would be to add gogoproto.no_stringer for now, with a custom String()

Reply:
Lol that's annoying. Is it just the `String()` method that is causing the panics or could there be other methods

Reply:
Actually, I remember coming across the same error statement in `x/authz` (https://github.com/cosmos/cosmos-sdk/pull/10944) whereby the querier was unable to parse the authz messages (I think to JSON but not sure)

Reply:
x/authz might have the same problem yeah, with nested anys.

Though I switched the test to use MarshalJSON, and it passes. I created #10965 to track that, so that we can move forward with this PR.